rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -1296,25 +1294,84 @@ public void onUncaughtException(Throwable e) {
         error(e.getMessage());
     }
 
-    private String parseScriptResult(String json) {
-        StringBuilder scriptResultStr = new StringBuilder();
-        JSONObject scriptResult = this.parseJSON(json).isObject();
+    private String parseAllScriptResults(String jsonString) {
+        JSONValue allScriptResultsJson = this.parseJSON(jsonString);
+        JSONObject scriptResultJsonObject = allScriptResultsJson.isObject();
+        if (scriptResultJsonObject != null) {
+            return parseScriptResultAsJsonObject(scriptResultJsonObject);
+        } else {
+            return parseAllScriptResultsAsJsonArray(allScriptResultsJson);
+        }
+    }
 
-        JSONObject exception = scriptResult.get("exception").isObject();
-        while (exception != null && exception.get("cause") != null && exception.get("cause").isObject() != null) {
-            exception = exception.get("cause").isObject();
+    private String parseAllScriptResultsAsJsonArray(JSONValue allScriptResultsJson) {
+        StringBuilder allScriptResultsOutput = new StringBuilder();
+        JSONArray scriptResultJsonArray = allScriptResultsJson.isArray();
+
+        JSONValue scriptResultJson;
+        JSONObject scriptResultJsonObject;
+        for (int i = 0; i < scriptResultJsonArray.size(); i++) {
```

Comment:
We can change as below (but I'm not sure it works)
for ( JSONValue scriptResultJson : scriptResultJsonArray) {

Reply:
No it doesn't, unfortunately it is not iterable :-)